tests/application/controllers/forget-password.spec.ts

```diff
@@ -0,0 +1,192 @@
+import { ForgetPasswordController } from '@/application/controllers/forget-password'
+import { type EmailValidator } from '@/application/validation/protocols'
+import { ServerError } from '@/application/errors'
+import { type LoadGuardianByEmail } from '@/domain/use-cases/load-guardian-by-email'
+import { type TokenGenerator } from '@/data/protocols/recovery-password/token-generator'
+import { type EmailService } from '@/domain/use-cases'
+
+const makeEmailValidator = (): EmailValidator => {
+  class EmailValidatorStub implements EmailValidator {
+    isValid (email: string): boolean {
+      return true
+    }
+  }
+  return new EmailValidatorStub()
+}
+
+const makeLoadGuardianByEmail = (): LoadGuardianByEmail => {
+  class LoadGuardianByEmailStub implements LoadGuardianByEmail {
+    async load (email: string): Promise<LoadGuardianByEmail.Result> {
+      return {
```

Comment:
Maybe we need create `mocks` and `stubs` folders to separe tests of helpers, reuse and  mantain tests more clean.